cpp/src/arrow/type.h

```diff
@@ -29,6 +29,7 @@
 
 #include "arrow/type_fwd.h"  // IWYU pragma: export
 #include "arrow/util/checked_cast.h"
+#include "arrow/util/decimal.h"
```

Comment:
Adding this include will worsen compile times. You should be able to add a forward declaration to `type_fwd.h` if needed, instead.

Reply:
This depends on the above.

Reply:
Moved to type_fwd.h